pycontrails/models/aircraft_performance.py

```diff
@@ -4,34 +4,405 @@
 
 import abc
 import dataclasses
-from typing import Any, overload
+import warnings
+from typing import Any, NoReturn, overload
 
+import numpy as np
+import numpy.typing as npt
+
+from pycontrails.core import flight
 from pycontrails.core.flight import Flight
 from pycontrails.core.met import MetDataset
-from pycontrails.core.models import Model, ModelParams
+from pycontrails.core.models import Model, ModelParams, interpolate_met
 from pycontrails.core.vector import GeoVectorDataset
+from pycontrails.physics import jet
+
+#: Default load factor for aircraft performance models.
+DEFAULT_LOAD_FACTOR = 0.7
+
+
+@dataclasses.dataclass
+class AircraftPerformanceParams(ModelParams):
+    """Parameters for :class:`AircraftPerformance`."""
+
+    #: Whether to correct fuel flow to ensure it remains within
+    #: the operational limits of the aircraft type.
+    correct_fuel_flow: bool = True
+
+    #: The number of iterations used to calculate aircraft mass and fuel flow.
+    n_iter: int = 5
 
 
 class AircraftPerformance(Model):
     """
     Support for standardizing aircraft performance methodologies.
 
-    Currently just a container until additional models are implemented.
+    This class provides a :meth:`simulate_fuel_and_performance` method for
+    iteratively calculating aircraft mass and fuel flow rate. The implementing
```

Comment:
I'd add a new line before `. The implementing...`

Reply:
e4f68aa